### ISSUE TITLE
Errors while using Protocol classes from `typing_extensions`

### ISSUE THREAD

**Issue by demberto** [microsoft/pyright#3860]
**Describe the bug**
`Argument to class must be a base class` and `Base class type is unknown, obscuring type of derived class` when using / creating Protocol classes from `typing_extensions`.

**To Reproduce**
Create a protocol like this (generic in this case):

```py
try:
    from typing import Protocol
except ImportError:
    from typing_extensions import Protocol

G = TypeVar("G")
S = TypeVar("S")

class Property(Protocol[G, S]):
    def __get__(self, instance: Any, owner: Any = None) -> G: ...
    def __set__(self, instance: Any, value: S): ...
```

**Expected behavior**
It shouldn't give an error.

**VS Code extension or command-line**
VS Code - Pylance **v2022.8.30**

**Additional context**
In fact any protocol I create has these same two errors, whether it is generic or not. It seems I cannot create protocols at all.

**EDIT**: I know think this is a typeshed issue, so if I am correct please close this issue

**Comment by erictraut** [comment]
You are importing `Protocol` from two different libraries, `typing` and `typing_extensions`. A static type analyzer doesn't know which of these two will be used at runtime, hence the problem.

If your code needs to run on Python versions prior to 3.8, you should import `Protocol` from `typing_extensions`. If your code runs on Python 3.8 and newer, you should import `Protocol` from `typing`. There's no need to import from both.

**Comment by demberto** [comment]
> There's no need to import from both.

So instead of catching `ImportError`, I should instead use `sys.version_info` or something?

**Comment by erictraut** [comment]
What I mean is that you should either use the statement `from typing_extensions import Protocol` or `from typing import Protocol` depending on whether you require backward compatibility with <3.8. There's no need to conditionally import from one or the other — i.e. no need for a `try` or `if` statement.

**Comment by demberto** [comment]
> What I mean is that you should either use the statement `from typing_extensions import Protocol` or `from typing import Protocol` depending on whether you require backward compatibility with <3.8. There's no need to conditionally import from one or the other — i.e. no need for a `try` or `if` statement.

What if I want to use `Protocol` in a library targeting Python 3.6+?

**Comment by erictraut** [comment]
Ah, so this is library code rather than app code? Thanks for that clarification. Then I presume you're trying to avoid taking a dependency on `typing_extensions` if the library is being used by an app that is intended to run only on Python 3.8 and newer? That makes sense. In that case, then I recommend using an `if` statement that checks on the system version, like this:

```python
import sys

if sys.version_info >= (3, 8):
    from typing import Protocol
else:
    from typing_extensions import Protocol
```

Static analysis tools like pyright and mypy are able to statically evaluate `sys.version_info` checks.

**Comment by demberto** [comment]
Thanks this is exactly what I needed
